### PR TITLE
Fixes CSV-reader type inference for thousands separator and decimal point

### PR DESCRIPTION
**Pull request by elstehle**
This PR fixes https://github.com/rapidsai/cudf/issues/6655 
This PR also makes sure to respect a user-specified decimal point during type inference. I.e., when the decimal point is not '.', types are now correctly inferred.
Plus some minor doxygen fixes and style changes from camelCase to snake_case.

### CODE REVIEW

File: cpp/tests/io/csv_test.cpp [hunk 1]
```diff
@@ -1015,6 +1015,41 @@ TEST_F(CsvReaderTest, StringInference)
...
+  cudf_io::csv_reader_options in_opts =
+    cudf_io::csv_reader_options::builder(cudf_io::source_info{buffer.c_str(), buffer.size()})
+      .header(-1)
+      .thousands('`');
+  const auto result = cudf_io::read_csv(in_opts);
+
+  EXPECT_EQ(result.tbl->num_columns(), 3);
+  EXPECT_EQ(result.tbl->get_column(0).type().id(), cudf::type_id::INT64);
+  EXPECT_EQ(result.tbl->get_column(1).type().id(), cudf::type_id::INT64);
+  EXPECT_EQ(result.tbl->get_column(2).type().id(), cudf::type_id::FLOAT64);
```
Comment (rgsl888prabhu) [inline cpp/tests/io/csv_test.cpp:1030]: Add a one more check to validate parsed values with expected column for both tests

**Review by rgsl888prabhu — COMMENTED** [review]
minor comment, other than that looks good.

File: cpp/tests/io/csv_test.cpp (hunk 1 above)
Comment (vuule) [reply cpp/tests/io/csv_test.cpp:1030]: is ti possible to use pandas for validation here?

**Review by vuule — APPROVED** [review]
Looks good except for the existing comment on testing :+1:

File: cpp/tests/io/csv_test.cpp (hunk 1 above)
Comment (elstehle) [reply cpp/tests/io/csv_test.cpp]: Have added the check 👍

**Review by rgsl888prabhu — APPROVED** [review]

**Comment by vuule** [comment]
@elstehle please don't force-push. If I remember correctly, it messes with the comment traceability.

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**